interceptor/main.go

```diff
@@ -77,35 +79,20 @@ func main() {
 		os.Exit(1)
 	}
 
-	configMapsInterface := cl.CoreV1().ConfigMaps(servingCfg.CurrentNamespace)
-
 	waitFunc := newDeployReplicasForwardWaitFunc(lggr, deployCache)
 
 	lggr.Info("Interceptor starting")
 
 	q := queue.NewMemory()
-	routingTable := routing.NewTable()
 
-	// Create the informer of ConfigMap resource,
-	// the resynchronization period of the informer should be not less than 1s,
-	// refer to: https://github.com/kubernetes/client-go/blob/v0.22.2/tools/cache/shared_informer.go#L475
-	configMapInformer := k8s.NewInformerConfigMapUpdater(
-		lggr,
-		cl,
-		servingCfg.ConfigMapCacheRsyncPeriod,
-		servingCfg.CurrentNamespace,
-	)
-
-	lggr.Info(
-		"Fetching initial routing table",
-	)
-	if err := routing.GetTable(
-		ctx,
-		lggr,
-		configMapsInterface,
-		routingTable,
-		q,
-	); err != nil {
+	httpCl, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		lggr.Error(err, "creating new HTTP ClientSet")
+		os.Exit(1)
+	}
+	sharedInformerFactory := informers.NewSharedInformerFactory(httpCl, servingCfg.ConfigMapCacheRsyncPeriod)
+	routingTable, err := routing.NewTable(sharedInformerFactory, "")
```

Comment:
Should we use an env var for the namespace? I mean, maybe some user wants to limit the interceptor to a specific namespace (or maybe not, in this stage we can check all the namespaces)

Reply:
That was something I thought when writing the code but I wasn’t sure if we were supporting namespaces installs of the add-on. I’ll check the state of that and make a patch here if we should already support that. Otherwise, I think we can implement that on another PR.